Source/Core/DolphinQt2/Config/ControllersWindow.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QBoxLayout>
+#include <QCheckBox>
+#include <QComboBox>
+#include <QDialog>
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QVBoxLayout>
+
+#include <unordered_map>
+
+#include "Core/HW/SI/SI.h"
+#include "Core/HW/Wiimote.h"
+
+class ControllersWindow final : public QDialog
+{
+  Q_OBJECT
+public:
+  explicit ControllersWindow(QWidget* parent);
+private slots:
```

Comment:
There should be a newline after the contructor here, like what's done with the section below. Many of these technically don't need to be slots and can just be private functions using Qt 5's event syntax (see other comment about them).